### DISCUSSION TITLE
Is it possible to use values from form in uncontrolled mode as keys in react query?

### DISCUSSION THREAD

**Discussion by dmitry-ff** [mantinedev/mantine#6966] [Q&A] [answered]
Is it possible to use values from form in uncontrolled mode as keys in react query?
Or i need to use onValuesChange for it?

**Comment by rtivital** [comment, marked as answer]
Uncontrolled form does not trigger a rerender – react query keys will not be updated when values change. In this case, it is better to use controlled form mode.
